### ISSUE TITLE
Show complete variable field in landscape view

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#6687]
Of course the new calculator function reaches the limits of the UI.
@S-Bartfast already implemented it quite well with this scrolling variable field. 

However I could imagine the view being optimized in landscape mode:
Currently it will keep the formulas and calculations on the left half of the screen, whereas the right half of the screen is filled with the note field.
For a better overview I would like to see the variable field expanding over the full screen width allowing an easy overview by just turning your device into landscape. 
The note field could just start underneath.

Here is how it currently looks:
![screenshot_20170816-110840](https://user-images.githubusercontent.com/949669/29356753-92f46140-8275-11e7-9afc-b4500e4a1615.png)

**Comment by SammysHP** [comment]
On the other hand notes are helpful if you've copied the formula into it. But a bigger view of the variables would be nice as well...

**Comment by S-Bartfast** [comment]
I'm more than open to suggestions on how this in particular can be improved.

To begin with I did just have the exact same layout in both portrait and landscape but it did look a bit strange on a tablet.  In particular the 'digit-buttons' were very broad and the 'variables' section looks kind of strange when the expressions were short:
![image](https://user-images.githubusercontent.com/6573522/29361081-1b1072ae-82c5-11e7-884f-027690c2b9d9.png)

On a phone it's not too bad:
![image](https://user-images.githubusercontent.com/6573522/29361272-febd3ef6-82c5-11e7-91d8-c31bf8408c2b.png)

But on a tablet it's really not that good, which is why I implemented the side-notes view:
![image](https://user-images.githubusercontent.com/6573522/29362166-f4fa586e-82c9-11e7-947d-efd9d2385c04.png)
Do we have any details on how many people use tablets for caching?

I'll look into exclusively using the 'side-notes' layout on _tablets_ in landscape orientation.


By the way, something I spent a lot of time battling with was how to make those 'GridView's appear properly.  You may have noticed that when no 'free-variables' are used (ie. only 'equations') the GridView fills the entire width of the screen, but once a variable is used and two GridViews are rendered both shrink to 'wrap-contents' and a white border is placed around them both.  (the idea behind the white border was to hint at the fact that there was suddenly more information off the edge of the screen).  Although this switch between filling the screen and wrapping the content works well it wasn't my original intention but I couldn't find a way to achieve my original idea –which was to have the center section 'swipe' between the equations and the free-variables, each filling the entire width of the screen.  As it is there are still a few bugs with the current implementation but on the whole it does seem to work okay.

**Comment by S-Bartfast** [comment]
Okay, I've just created a Pull Request that addresses this.

The solution proposed is to only use the 'side-note' layout when the screen width is greater than 1200dp.

[This page](https://material.io/devices/) shows the dimensions of most of the popular screens on the market at present.  1200dp was chosen because although the standard layout starts to look a little strange it's still okay on the Nexus 9 (1024dp):
![image](https://user-images.githubusercontent.com/6573522/29370439-799cf03a-82e4-11e7-8a55-d07c8f79fb8c.png)

But really doesn't look so good on the Nexus 10 (1280dp):
![image](https://user-images.githubusercontent.com/6573522/29361081-1b1072ae-82c5-11e7-884f-027690c2b9d9.png)

However the 'side-note' layout definitely does look fine on the Nexus 10:
![image](https://user-images.githubusercontent.com/6573522/29370629-1bfe934c-82e5-11e7-80b6-3b6cde692275.png)

**Comment by Lineflyer** [comment]
IMHO that is a good compromise.
Once it is merged I can test the look on different devices and we might change the dp limit later on if necessary,

**Comment by SammysHP** [comment]
> Once it is merged

Every PR built on our CI server should contain a signed APK somewhere, right?

**Comment by Lineflyer** [comment]
> Every PR built on our CI server should contain a signed APK somewhere, right?

Yes, but I tend to use this option only if a change is too experimental to be merged but still should undergo testing. Else I prefer testing with nightlies.

**Comment by Lineflyer** [comment]
Assigned for retest.

**Comment by Lineflyer** [comment]
Tested with 2017.08.21-NB:
- Samsung S7 (below limit) is showing landscape and portrait in same style.
- Samsung Galaxy Tab S2 is also showing the same style in portrait and landscape. Right now I did not find the dp specification for this device. I would have assumes it is above the limit, but I might be wrong.

**Comment by S-Bartfast** [comment]
> Right now I did not find the dp specification for this device

Try checking it with [What's my Size?](https://play.google.com/store/apps/details?id=nl.qbus.sizemeup&hl=en&rdid=nl.qbus.sizemeup):
![image](https://user-images.githubusercontent.com/6573522/29544985-cf8d957c-872a-11e7-8916-de4c28ec2055.png)

**Comment by Lineflyer** [comment]
Sorry for letting you wait:

My tablet has this size:
![screenshot_20171024-210111](https://user-images.githubusercontent.com/949669/31962749-b75ff520-b8fe-11e7-9d60-7e30e0035078.png)

From my impression on my tablet it would better fit with the enhanced layout. 
IMHO we could already start at >1000dp to use the other layout.

**Comment by S-Bartfast** [comment]
Hmm, 1024dp.  That's the same as the Nexus 9 (which was actually the example used to determine that 1024 limit).

I have a lot of screen shots but I believe the two I was comparing at the time I decided 1024 could stay with the 'standard' layout was these two:
![image](https://user-images.githubusercontent.com/6573522/32046346-59d817c4-ba8a-11e7-8048-4fcda7f87c6d.png)

![image](https://user-images.githubusercontent.com/6573522/32046353-5eda0ffc-ba8a-11e7-9b45-651f49b38c33.png)

Of course I can set the 'switching' resolution to be lower if you think the side-by-side layout is better for 1024 dp displays but, if you recall, the request of this ticket was actually to remove the side layout altogether so I was trying to find the limit as to where the 'standard' layout started to look silly rather than where the side-by-side layout started to look reasonable.  As it was 1024 dp is on the cusp and at the time I decided to err on the side of favouring the standard layout but can easily change it if you wish.

Just as a note...
Samsung do have a tendency to break from the norm a little when they come to describe how many dp their displays should represent.  If I recall correctly they tend to under state their dp value (ie. make icons and things appear bigger on their display than others) and if that's how they want things displayed on their devices they are of course most welcome but it does mean we should resist tailoring the UX to suit Samsung's devices in particular.  You know, if Samsung want graphics displayed a little larger on their displays than we should have them display a little larger rather than try to compensate, but I don't think that's actually an issue here.  Just mentioning so you know ;)
